### DISCUSSION TITLE
Non-Hue device in Philips Hue app -> Raspberrymatic -> Hue Kopplung -> HAP-Homematic -> Apple Home (mit Problemen)

### DISCUSSION THREAD

**Discussion by bgewehr** [thkl/hap-homematic#596] [Q&A]
Hallo,

da die Philips Hue bridge ja non-hue devices einbinden kann, aber nicht in Apple Home überträgt, kam mir die Idee, dies mit Hilfe der Hue Kopplung in Raspberrymatic und HAP-Homematic zu tun, denn die per Hue Kopplung in Homematic ergänzten Devices lassen sich dort leicht hinzufügen. Leider lassen sich die Devices schon in Homematic nicht einwandfrei bedienen, über HAP-Homematic in Apple Home dann fast gar nicht mehr.

Aber der Ansatz ist doch interessant, um die Leuchten und Steckdosen von z. B. INNR und Co auch in das Apple Home einzubeziehen, ohne dafür wieder einen Raspi mit Homebridge und Zigbee2MQTT zu benötigen.

Konkret habe ich eine INNR FL120C Lightstrip und eine INNR smart plug SP220. In der Hue App lassen sich beide noch einwandfrei steuern. In Homematic funktioniert die Schaltung des Plugs, aber der Status wird immer auf "Aus" dargestellt. Der Lightstrip stellt in der nighly build von Raspberrymatic vom 28.12.22 die Farbauswahl zwar wieder ohne JS-Fehler dar, aber die Farben stimmen nicht, außerdem steht da nicht rot/grün/blau, sondern Sättigung usw.. Wird sicher bald gefixt.

In HAP-Homematic aber klappt fast nichts mehr. Die Geräte lassen sich leicht hinzufügen und erscheinen auch sofort mit den richtigen Kontrollen in Apple Home. Der Strip lässt sich dimmen, das ist aber auch schon alles. Keine weitere Funktion möglich.

Ich würde gern helfen, der Sache auf den Grund zu gehen, allerdings bräuchte ich etwas Anleitung, wie ich die Fehleranalyse angehen könnte. Wie könnte ich anfangen?

**Comment by jb-home** [comment]
Ich würde HAP im debug Modus starten.
Dann gezielt Aktionen starten und sich entsprechend dazu das LOG anschauen, ob da ggf. falsche Daten übertragen werden.
Kann gut sein, dass die Farbwerte andere Min & Max Werte haben, wodurch das Mischen der Farben dann natürlich durcheinander kommt.
Wenn es auf der CCU aber schon nicht richtig funktioniert, macht ein Debuggen auf HAP eigentlich nicht wirklich viel Sinn. Du wirst es wenn nur soweit schaffen, das die Geräte sich in HAP so verhalten wie in der CCU.

**Comment by bgewehr** [comment]
Mit dem Tipp von hier:
https://homematic-forum.de/forum/viewtopic.php?t=62849
bin ich einen Schritt weiter gekommen, denn im Vergleich zu der sehr rudimentären Realisierung in der Hue-Kopplung macht das Hue Plugin einen besseren Eindruck.

Habe die Kopplung und alle darüber aufgenommenen Geräte entfernt. CUX und Hue Plugin aktuell installiert und die Bridge hinzugefügt, dann die zwei INNR Geräte als CUX-Switch und CUX-RGBW device hinzugefügt, diese im Posteingang der CCU gefunden und zur Geräteliste hinzugefügt.

Zwischenstand:
Der Plug schaltet von Homematic und Apple Home aus korrekt.
Der Lightstrip lässt sich in Homematic ein- und ausschalten, dimmen und die Farbe wählen - allerdings in Apple Home eher mit Zufallswirkung. Witzig und vorher noch nie so erlebt: Wenn ich in Home den Dimmer hochschiebe, ändert sich die Farbe statt der Helligkeit. Ein/aus geht aber.

Hier nun dieselbe Frage: Welche Komponente macht hier was falsch? Wo lohnt es sich, zu testen?

**Comment by jb-home** [comment]
HAP im Debug Modus starten. Und dann die LOGs anschauen.
Bei HUE ist das mit den Farben etwas anders als nur RGB. Da kommen noch andere Werte hinzu, die mir gerade vom Namen her nicht einfallen wollen.
